taigapy/client_v3.py

```diff
@@ -312,7 +314,86 @@ def _get(self, datafile_id: str) -> pd.DataFrame:
 
         return result
 
-    def download_to_cache(self, datafile_id: str, requested_format: Union[LocalFormat, str]) -> str:
+    def upload_to_gcs(
+        self,
+        data_file_taiga_id: str,
+        requested_format: LocalFormat,
+        dest_gcs_path_for_file: str,
+    ) -> bool:
+        """Upload a Taiga datafile to a specified location in Google Cloud Storage.
+
+        The service account taiga-892@cds-logging.iam.gserviceaccount.com must have
+        storage.buckets.create access for this request.
+
+        Arguments:
+            `data_file_taiga_id` -- Taiga ID in the form dataset_permaname.dataset_version/datafile_name
+            `requested_format` -- The format of the file you want to upload
+            `dest_gcs_path_for_file` -- Google Storage path to upload to, in the form 'gs://bucket:file_obj_name'
+
+        Returns:
+            bool -- Whether the file was successfully uploaded
+        """
+
+        def parse_gcs_path(gcs_path: str) -> Tuple[str, str]:
+            gcs_path_error = ValueError(
+                "Invalid GCS path. '{}' is not in the form 'gs://bucket_name/object_name'".format(
+                    gcs_path
+                )
+            )
+            if not gcs_path.startswith("gs://"):
+                raise gcs_path_error
+            # remove prefix
+            gcs_path = gcs_path.replace("gs://", "")
+
+            if "/" not in gcs_path:
+                raise gcs_path_error
+
+            bucket_name, file_object_name = gcs_path.split("/", 1)
+            return (
+                bucket_name,
+                file_object_name,
+            )  # bucket_name is dest_bucket (w/o gs:// prefix) object_name is dest_path
+
+        def get_bucket(bucket_name: str) -> storage.Bucket:
+            client = storage.Client()
+            try:
+                bucket = client.get_bucket(bucket_name)
+            except gcs_exceptions.Forbidden as e:
+                raise ValueError(
+                    "taiga-892@cds-logging.iam.gserviceaccount.com does not have storage.buckets.get access to bucket: {}".format(
```

Comment:
This exception isn't true. I'd just let the Forbidden exception be thrown because it'll have the correct information in it. (I don't believe you're using this service account, correct?)

Reply:
Oh! I just copied this error comment from the Taiga library. I'll just raise that exception as is then